### ISSUE TITLE
`separate_rows` fails silently if there are no other columns

### ISSUE THREAD

**Issue by rdboyes** [TidierOrg/TidierData.jl#125]
If we have a dataframe: 

```
df = DataFrame(Column1 =[
"7 6 4 2 1",
"1 2 7 8 9",
"9 7 6 2 1",
"1 3 2 4 5",
"8 6 4 4 1",
"1 3 6 7 9"]
)
```

And try to separate it into rows: 

```
@separate_rows(df, Column1, " ")
```

The returned `DataFrame` is unchanged and no error is thrown. The function works as expected if another Column is present, e.g:

```
@chain df begin
    @mutate(id = 1:6)
    @separate_rows(Column1, " ")
end
```
returns
```
30×2 DataFrame
 Row │ Column1    id
     │ SubStrin…  Int64
─────┼──────────────────
   1 │ 7              1
   2 │ 6              1
  ⋮  │     ⋮        ⋮
  29 │ 7              6
  30 │ 9              6
         26 rows omitted
```

**Comment by drizk1** [comment]
Never thought to test for that when I wrote it. Happy to fix it

**Comment by rdboyes** [comment]
Advent of code makes you write weird code! It's not a common way to use the function but it took me a minute to figure out why it was failing

**Comment by rdboyes** [comment]
I think the problem may have something to do with InlineStrings - when I tested it, I was reading that little DataFrame from a file, and I can replicate the behaviour if I declare it as a String15:
![image](https://github.com/user-attachments/assets/8bc81d4a-869d-4876-b65c-2093922e6329)

**Comment by drizk1** [comment]
wow unique situation, but good edge case to know about. i had no idea.
```
if ismissing(value) || typeof(value) != String
#changed to 
if ismissing(value) || !(value isa AbstractString)
```
and it now should work works 
```
julia> df = DataFrame(Column1 = String15[
       "7 6 4 2 1",
       "1 2 7 8 9",
       "9 7 6 2 1",
       "1 3 2 4 5",
       "8 6 4 4 1",
       "1 3 6 7 9"]
       );

julia> @separate_rows(df, Column1, " ")
30×1 DataFrame
 Row │ Column1   
     │ SubStrin… 
─────┼───────────
   1 │ 7
   2 │ 6
   3 │ 4
   4 │ 2
   5 │ 1
```

**Comment by kdpsingh** [comment]
Resolved in #127.
